contracts/04-tamagotchi-shop/src/lib.rs

```diff
@@ -18,7 +20,15 @@ pub struct Tamagotchi {
     pub slept: u64,
     pub slept_block: u64,
     pub approved_account: Option<ActorId>,
+    pub ft_contract_id: ActorId,
+    pub transaction_id: TransactionId,
+    pub approve_transaction: Option<(TransactionId, ActorId, u128)>,
 }
+pub type TransactionId = u64;
+pub type AttributeId = u32;
+pub type Price = u128;
+pub type TamagotchiId = ActorId;
```

Comment:
No need to double the code. It is better to reuse it from the `io ` crate.

Reply:
I got.